### ISSUE TITLE
py.test can not run specific test from command line if setup.cfg is present

### ISSUE THREAD

**Issue by cpaulik** [pytest-dev/pytest#868]
I have a setup.cfg in the root directory of my project which has the pytest section:

```
[pytest]
# Options for py.test:
# Specify command line options as you would do when invoking py.test directly.
# e.g. --cov-report html (or xml) for html/xml output or --junitxml junit.xml
# in order to write a coverage file that can be read by Jenkins.
addopts = tests
    --cov smecv --cov-report term-missing
    --verbose

[aliases]
test = pytest
```

When I now try to run a specific test using e.g.:

```shell
py.test -x -s tests/input/test_cdf_parameters.py::test_amsr2_read_ts
```

all the tests are run and not just the one I selected. 
Renaming or removing setup.cfg solves the problem.

**Comment by nicoddemus** [comment]
The problem is your `addopts` line, which is including the `tests` folder for all `py.test` invocations. 

If the reason for `tests` being in `addopts` is to speed up collection, you can use `norecursedirs` instead to skip directories which that don't contain tests.

```
Set the directory basename patterns to avoid when recursing for test discovery.
```

Alternatively, the new `testpaths` option is more convenient but that is available in pytest `2.8` only (unreleased yet):

```
- New `testpaths` ini option: list of directories to search for tests
  when executing pytest from the root directory. This can be used
  to speed up test collection when a project has well specified directories
  for tests, being usually more practical than configuring norecursedirs for
  all directories that do not contain tests.
```

**Comment by cpaulik** [comment]
Thank you, that solves the problem.
